schema.cc

```diff
@@ -101,11 +102,21 @@ std::ostream& operator<<(std::ostream& os, ordinal_column_id id)
     return os << static_cast<column_count_type>(id);
 }
 
+std::unique_ptr<dht::i_partitioner> default_partitioner;
+
+void schema::set_default_partitioner(const sstring& class_name, unsigned ignore_msb)
+{
+    default_partitioner = dht::make_partitioner(class_name, smp::count, ignore_msb);
+}
+
 const dht::i_partitioner& schema::get_partitioner() const {
     if (_raw._partitioner) {
         return *_raw._partitioner;
     }
-    return dht::global_partitioner();
+    if (!default_partitioner) {
+        default_partitioner = std::make_unique<dht::murmur3_partitioner>(smp::count, 12);
+    }
+    return *default_partitioner;
 }
 
```

Comment:
It looks like you have two ways to set the default partitioner, one from set_default_partitioner() and one here. This means that schemas created before set_default_partitioner() is called may have the wrong partitioner.

The code here is racy. If multiple shards hit it at the same time, then each will try to write to the default_partitioner global.

Reply:
This is removed in the following patch. But you're right. It used to be the same before though in global_partitioner. It's a fair point though. I need to make partitioners cache thread-local otherwise it's also racy.